pkg/chains/formats/intotoite6/extract/extract.go

```diff
@@ -105,3 +106,15 @@ func SubjectDigests(obj objects.TektonObject, logger *zap.SugaredLogger) []intot
 	})
 	return subjects
 }
+
+// SubjectsToURIs converts intoto subjects to a slice of string URIs in the format
+// of "NAME" + "@" + "ALGORITHM" + ":" + "DIGEST"
+func SubjectsToURIs(subjects []intoto.Subject) []string {
+	result := []string{}
+	for _, s := range subjects {
+		for algo, digest := range s.Digest {
+			result = append(result, fmt.Sprintf("%s@%s:%s", s.Name, algo, digest))
```

Comment:
Happy to add the tests!
Regarding whether return error or empty slice, my thought is that if the intoto statement has empty subject or subject field is missing, it just indicates the build does not produce any artifact/output rather than a conversion error. As such we may just want to return empty slice as well? wdyt?
When it comes storage, 
- grafeas will not create any occurrences for the statement with empty subjects, and just log the information saying no occurrences were created in the controller log.
- [OCI storage](https://github.com/tektoncd/chains/blob/43615bdd7efbae5d7245d8f22bba6718f0ecc3bd/pkg/chains/storage/oci/oci.go#L94-L106) seems to have similar behaviour. And we may want to refactor OCI storage later to use this helper function as well.

Reply:
[tests](https://github.com/tektoncd/chains/pull/590/files#diff-852530124ab21d01f48bb7e3a76cb200a0c23d5e876844fb45dec65c044530df) are added. PTAL. Thank you! @wlynch

Reply:
To make the functionality reusable of being able to extract artifact URIs, I moved the `RetrieveAllArtifactURIs` function from grafeas package to this extract package so that it can be reused in oci or any other places as needed. The [tests](https://github.com/tektoncd/chains/pull/590/files#diff-852530124ab21d01f48bb7e3a76cb200a0c23d5e876844fb45dec65c044530df) are added.

PTAL. Thank you @wlynch !!

Reply:
👍  empty slice is fine with me - make sure the behavior is documented (e.g. what's considered valid)!